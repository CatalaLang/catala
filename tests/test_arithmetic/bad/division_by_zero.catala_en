## `Division_by_zero` exception management

### with integers

```catala
declaration scope Int:
  context i content integer

scope Int:
  definition i equals 1 / 0
```

### with decimals

```catala
declaration scope Dec:
  context i content decimal

scope Dec:
  definition i equals 1. /. 0.
```

### with money

```catala
declaration scope Money:
  context i content decimal

scope Money:
  definition i equals $10.0 /$ $0.0
```

### with duration

```catala
declaration scope Duration:
  context i content decimal

scope Duration:
  definition i equals 10 day /^ 0 day
```
