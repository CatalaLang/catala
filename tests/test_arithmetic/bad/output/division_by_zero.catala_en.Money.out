[ERROR] division by zero at runtime
[ERROR] 
[ERROR] The division operator:
[ERROR]   --> test_arithmetic/bad/division_by_zero.catala_en
[ERROR]    | 
[ERROR] 30 |   definition i equals $10.0 /$ $0.0
[ERROR]    |                             ^^
[ERROR]    + `Division_by_zero` exception management
[ERROR]    +-+ with money
[ERROR] 
[ERROR] The null denominator:
[ERROR]   --> test_arithmetic/bad/division_by_zero.catala_en
[ERROR]    | 
[ERROR] 30 |   definition i equals $10.0 /$ $0.0
[ERROR]    |                                ^^^^
[ERROR]    + `Division_by_zero` exception management
[ERROR]    +-+ with money
